README.md

```diff
@@ -1,114 +1,147 @@
+Please note that if you need to check a Node.JS script process cpu and memory usage, you can use [`process.cpuUsage`](https://nodejs.org/api/process.html#process_process_cpuusage_previousvalue) and [`process.memoryUsage`](https://nodejs.org/api/process.html#process_process_memoryusage) since node v6.1.0. This script remain useful when you have no control over the remote script, or if the process is not a Node.JS process.
 
-## API
 
-```javascript
+## Usage
+
+```js
 var pusage = require('pidusage')
 
 // Compute statistics every second:
-
 setInterval(function () {
-    pusage.stat(process.pid, function (err, stat) {
-
-	expect(err).to.be.null
-	expect(stat).to.be.an('object')
-	expect(stat).to.have.property('cpu')
-	expect(stat).to.have.property('memory')
-
-	console.log('Pcpu: %s', stat.cpu)
-	console.log('Mem: %s', stat.memory) //those are bytes
-
-    })
+  pusage.stat(process.pid, function (err, stat) {
+    console.log(stat)
+    // => {
+    //   cpu: 10.0,            // percentage (it may happen to be greater than 100%)
+    //   memory: 357306368,    // bytes
+    //   ppid: 312,            // PPID
+    //   pid: 727,             // PID
+    //   ctime: 867000,        // ms user + system time
+    //   elapsed: 6650000,     // ms since the start of the process
+    //   timestamp: 864000000  // ms since epoch
+    // }
+  })
 }, 1000)
 
-```
-
-When you're done with the given `pid`, you may want to clear `pidusage` history (it only keeps the last stat values):
-
-```
-pusage.unmonitor(process.pid);
-```
-
-The `stat` object will contain the following:
-
-```
-- `cpu` cpu percent
-- `memory` memory bytes
-- `time` elapsed time since started
-- `start` Date when process was started
-```
-
-Pidusage also supports an array of pids:
-
-```javascript
-var pusage = require('pidusage')
-
-pusage.stat([0,1,2], function (err, stats) {
-  // stats is an array of statistics objects
+// It supports also multiple pids
+pusage.stat([727, 1234], function (err, stat) {
+  console.log(stat)
+  // => {
+  //   727: {
+  //     cpu: 10.0,            // percentage
+  //     memory: 357306368,    // bytes
+  //     ppid: 312,            // PPID
+  //     pid: 727,             // PID
+  //     ctime: 867000,        // ms user + system time
+  //     elapsed: 6650000,     // ms since the start of the process
+  //     timestamp: 864000000  // ms since epoch
+  //   },
+  //   1234: {
+  //     cpu: 0.1,             // percentage
+  //     memory: 3846144,      // bytes
+  //     ppid: 727,            // PPID
+  //     pid: 1234,            // PID
+  //     ctime: 0,             // ms user + system time
+  //     elapsed: 20000,       // ms since the start of the process
+  //     timestamp: 864000000  // ms since epoch
+  //   }
+  // }
```

Comment:
Actually is a bit complicated than we thought.

When you loop over the results you don't know in which index you need to place the result in the output array.
In order to know that you need to build an hashmap and this would be too expensive and useful on our side.

Eg:
The user gives you: `pids = [123, 412, 124]`
You need to create:
`map = {'123':0, '412':1, '124':2}`

Then you can do
`output = []`
`output[map[pid]] = stats`

Since we need to create the map there's no reason to prefer the array solution that would also be slower for us.

Reply:
Okay I agree with that, also makes things easier in the results parsing!